modules/core/src/effects/lighting/sunlight.js

```diff
@@ -0,0 +1,138 @@
+import {DirectionalLight} from '@luma.gl/core';
+
+// sun position calculations are based on http://aa.quae.nl/en/reken/zonpositie.html formulas
+// based on https://github.com/mourner/suncalc/blob/master/suncalc.js
+const PI = Math.PI;
+const sin = Math.sin;
+const cos = Math.cos;
+const tan = Math.tan;
+const asin = Math.asin;
+const atan = Math.atan2;
+const rad = PI / 180;
+
+const DAY_IN_MS = 1000 * 60 * 60 * 24;
+const J1970 = 2440588;
+const J2000 = 2451545;
+const e = rad * 23.4397; // obliquity of the Earth
+
+function toJulian(timestamp) {
+  return timestamp / DAY_IN_MS - 0.5 + J1970;
+}
+
+function toDays(timestamp) {
+  return toJulian(timestamp) - J2000;
+}
+
+function getRightAscension(eclipticLongitude, b) {
+  const lambda = eclipticLongitude;
+  return atan(sin(lambda) * cos(e) - tan(b) * sin(e), cos(lambda));
+}
+
+function getDeclination(eclipticLongitude, b) {
+  const lambda = eclipticLongitude;
+  return asin(sin(b) * cos(e) + cos(b) * sin(e) * sin(lambda));
+}
+
+function getAzimuth(hourAngle, latitudeInRadians, declination) {
+  const H = hourAngle;
+  const phi = latitudeInRadians;
+  const delta = declination;
+  return atan(sin(H), cos(H) * sin(phi) - tan(delta) * cos(phi));
+}
+
+function getAltitude(hourAngle, latitudeInRadians, declination) {
+  const H = hourAngle;
+  const phi = latitudeInRadians;
+  const delta = declination;
```

Comment:
Why are these variables needed?

Reply:
I feel it is better to match the arguments as the wiki doc. These symbols are generally used in calculations.